app/server.js

```diff
@@ -1,13 +1,184 @@
 var http = require('http');
 var fs = require('fs');
 var finalHandler = require('finalhandler');
-var queryString = require('querystring');
+// var queryString = require('querystring');
 var Router = require('router');
-var bodyParser   = require('body-parser');
+var bodyParser = require('body-parser');
 var uid = require('rand-token').uid;
 
-const PORT = 3001;
+let chai = require('chai');
+let chaiHttp = require('chai-http');
+chai.use(chaiHttp);
 
-http.createServer(function (request, response) {
+// State holding variables
+var PORT = 3500;
+var brands = [];
+var products = [];
+var users = [];
+var accessTokens = [];
+const TOKEN_VALIDITY_TIMEOUT = 15 * 60 * 1000; // 15 minutes
+
+chai.request('http://localhost:3500');
+should = chai.should();
+chai.use(chaiHttp);  
+
+// Server
+const server = http.createServer((req, res) => {
+  res.writeHead(200);
+  router(req, res, finalHandler(req, res));
+});
+
+// Setup Router
+let router = Router();
+router.use(bodyParser.json());
+
+server.listen(PORT, err => {
+  if (err) throw err;
+  brands = JSON.parse(fs.readFileSync('initial-data/brands.json', 'utf-8'));
+  products = JSON.parse(fs.readFileSync('initial-data/products.json', 'utf-8'));
+  users = JSON.parse(fs.readFileSync('initial-data/users.json', 'utf-8'));
+});
+
+
+// 1. Requests for brands (passed):
+router.get('/brands', (req, res) => {
+  res.writeHead(200, { "Content-Type": "application/json" });
+  return res.end(JSON.stringify(brands));
+})
+
+// 2. Requests for products (passed):
+router.get('/products', (req, res) => {
+  res.writeHead(200, { "Content-Type": "application/json" });
+  return res.end(JSON.stringify(products));
+})
+
+// 3. Requests for a specific product (passed):
+router.get('/products/:categoryId', (req, res) => {
+  const productId = req.params.categoryId;
```

Comment:
What if a bad `categoryId` is passed here? We'd probably want to send back a 404/401.